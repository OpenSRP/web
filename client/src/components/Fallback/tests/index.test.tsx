import { mount, shallow } from 'enzyme';
import React from 'react';
import Fallback from '..';

describe('/components/Fallback', () => {
  it('renders without crasshing', () => {
    shallow(<Fallback />);
  });

  it('render correctly', () => {
    const wrapper = mount(<Fallback />);
    expect(wrapper.find('Result')).toHaveLength(1);
    expect(wrapper.find('Result').prop('status')).toBe('error');
    expect(wrapper.find('Result').prop('title')).toBe('An Error Occurred');
    expect(wrapper.find('Result').prop('subTitle')).toBe(
      'There has been an error. Itâ€™s been reported to the site administrators via email and should be fixed shortly. Thanks for your patience.'
    );
  });
});
