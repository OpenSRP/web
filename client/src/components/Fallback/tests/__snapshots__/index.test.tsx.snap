// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/components/Fallback render correctly 1`] = `
<Fallback>
  <Result
    extra={
      Array [
        <Button
          block={false}
          ghost={false}
          htmlType="button"
          loading={false}
          onClick={[Function]}
          type="primary"
        >
          Back Home
        </Button>,
      ]
    }
    status="error"
    subTitle="There has been an error. It’s been reported to the site administrators via email and should be fixed shortly. Thanks for your patience."
    title="An Error Occurred"
  >
    <div
      className="ant-result ant-result-error"
    >
      <div
        className="ant-result-icon"
      >
        <ForwardRef(CloseCircleFilled)>
          <AntdIcon
            icon={
              Object {
                "icon": Object {
                  "attrs": Object {
                    "focusable": "false",
                    "viewBox": "64 64 896 896",
                  },
                  "children": Array [
                    Object {
                      "attrs": Object {
                        "d": "M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z",
                      },
                      "tag": "path",
                    },
                  ],
                  "tag": "svg",
                },
                "name": "close-circle",
                "theme": "filled",
              }
            }
          >
            <span
              aria-label="close-circle"
              className="anticon anticon-close-circle"
              role="img"
            >
              <IconReact
                className=""
                icon={
                  Object {
                    "icon": Object {
                      "attrs": Object {
                        "focusable": "false",
                        "viewBox": "64 64 896 896",
                      },
                      "children": Array [
                        Object {
                          "attrs": Object {
                            "d": "M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z",
                          },
                          "tag": "path",
                        },
                      ],
                      "tag": "svg",
                    },
                    "name": "close-circle",
                    "theme": "filled",
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="close-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  key="svg-close-circle"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                    key="svg-close-circle-svg-0"
                  />
                </svg>
              </IconReact>
            </span>
          </AntdIcon>
        </ForwardRef(CloseCircleFilled)>
      </div>
      <div
        className="ant-result-title"
      >
        An Error Occurred
      </div>
      <div
        className="ant-result-subtitle"
      >
        There has been an error. It’s been reported to the site administrators via email and should be fixed shortly. Thanks for your patience.
      </div>
      <div
        className="ant-result-extra"
      >
        <Button
          block={false}
          ghost={false}
          htmlType="button"
          key="error"
          loading={false}
          onClick={[Function]}
          type="primary"
        >
          <Wave>
            <button
              className="ant-btn ant-btn-primary"
              onClick={[Function]}
              type="button"
            >
              <LoadingIcon
                existIcon={false}
                loading={false}
                prefixCls="ant-btn"
              >
                <ForwardRef
                  motionName="ant-btn-loading-icon-motion"
                  onAppearActive={[Function]}
                  onAppearStart={[Function]}
                  onEnterActive={[Function]}
                  onEnterStart={[Function]}
                  onLeaveActive={[Function]}
                  onLeaveStart={[Function]}
                  removeOnLeave={true}
                  visible={false}
                >
                  <CSSMotion
                    internalRef={null}
                    motionAppear={true}
                    motionEnter={true}
                    motionLeave={true}
                    motionName="ant-btn-loading-icon-motion"
                    onAppearActive={[Function]}
                    onAppearStart={[Function]}
                    onEnterActive={[Function]}
                    onEnterStart={[Function]}
                    onLeaveActive={[Function]}
                    onLeaveStart={[Function]}
                    removeOnLeave={true}
                    visible={false}
                  />
                </ForwardRef>
              </LoadingIcon>
              <span
                key=".0"
              >
                Back Home
              </span>
            </button>
          </Wave>
        </Button>
      </div>
    </div>
  </Result>
</Fallback>
`;
